contracts/WrappedCollateralToken.sol

```diff
@@ -37,21 +37,27 @@ contract WrappedCollateralToken is IWrappedCollateralToken, ERC20Wrapper, ERC20P
         emit CapSet(newCap);
     }
 
-    function depositFor(address account, uint256 amount) public virtual override returns (bool) {
-        if (totalSupply() + amount > cap) {
-            revert ExceedsCap();
-        }
-        if (balanceOf(account) + amount > maxBalance) {
-            revert ExceedsMaxBalance();
-        }
-        return ERC20Wrapper.depositFor(account, amount);
-    }
-
     function decimals() public view virtual override(ERC20, ERC20Wrapper) returns (uint8) {
         return ERC20Wrapper.decimals();
     }
 
     function recover(address account) external override onlyOwner returns (uint256) {
         return _recover(account);
     }
+
+    function _mint(address account, uint256 amount) internal virtual override {
+        ERC20._mint(account, amount);
+        if (totalSupply() > cap) {
+            revert ExceedsCap();
+        }
+    }
+
+    /// @dev This is called by OZ ERC20 for mint and all transfers after the actual transfer.
+    /// In case of burn it calls with to set to 0. Implementing balance check here is sufficient.
+    function _afterTokenTransfer(address from, address to, uint256 amount) internal virtual override {
```

Comment:
Nice, because of this, no one will have the possibility to buy with different accounts, and then transfer to one account.

Reply:
yup